src/Compilers/CSharp/Portable/Lowering/BoundTreeToDifferentEnclosingContextRewriter.cs

```diff
@@ -142,6 +170,44 @@ protected BoundBlock VisitBlock(BoundBlock node, bool removeInstrumentation)
                 VisitType(node.Type));
         }
 
+        [return: NotNullIfNotNull(nameof(property))]
+        public override PropertySymbol? VisitPropertySymbol(PropertySymbol? property)
+        {
+            if (property is null)
+            {
+                return null;
+            }
+
+            if (property.ContainingType.IsAnonymousType)
+            {
+                //  Property of an anonymous type
```

Comment:
>//  Property of an anonymous type [](http://example.com/codeflow?start=16&length=33)

It doesn't look like this code path should be reachable. consider leaving just ```throw ExceptionUtilities.Unreachable();``` in this branch.

Reply:
Otherwise we should add tests for this code path.

Reply:
I think we should keep the `if` and throw, or we should at least have an assert.